clients/src/test/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegateTest.java

```diff
@@ -212,9 +213,27 @@ public void testPropagateMetadataError() {
         AuthenticationException authException = new AuthenticationException("Test Auth Exception");
         doThrow(authException).when(metadata).maybeThrowAnyException();
 
+        NetworkClientDelegate networkClientDelegate = newNetworkClientDelegate(false);
+        assertTrue(networkClientDelegate.getAndClearMetadataError().isEmpty());
+        networkClientDelegate.poll(0, time.milliseconds());
+        
+        assertTrue(networkClientDelegate.getAndClearMetadataError().isPresent());
```

Comment:
this will clear the error so the assertions within the ifPresent will never run right? (maybe we take the Optional result of .getAndClear, check that optional is present, and then the 2 asserts on the value?)

Reply:
You are right, I missed that we will clear the metadata error.